src/autopas/LogicHandler.h

```diff
@@ -1063,18 +1191,182 @@ void LogicHandler<Particle>::remainderHelperBufferHaloBuffer(
+
+      container.forEachInRegion(
+          [&](auto &p3) {
+            const auto lockCoords = static_cast_copy_array<size_t>((p3.getR() - haloBoxMin) * interactionLengthInv);
+            // no newton3 here due to race conditions
+            if (i < numOwnedBufferParticles) f->AoSFunctor(p1, p2, p3, false);
+            // no need to calculate force enacted on a halo
+            if (not p3.isHalo() and i < j) {
+              const std::lock_guard<std::mutex> lock(*_spacialLocks[lockCoords[0]][lockCoords[1]][lockCoords[2]]);
+              f->AoSFunctor(p3, p1, p2, false);
+            }
+          },
+          min, max, IteratorBehavior::ownedOrHalo);
+    }
+  }
+
+#if SPDLOG_ACTIVE_LEVEL <= SPDLOG_LEVEL_TRACE
+  timerBufferBufferContainer.stop();
+  timerBufferContainerContainer.start();
+#endif
+
+  // Step 3: Triwise interactions of 1 buffer particle and 2 container particles
+  // todo: parallelize without race conditions - https://github.com/AutoPas/AutoPas/issues/904
+  // AUTOPAS_OPENMP(parallel for shared(bufferParticles))
+  for (auto i = 0; i < numTotal; ++i) {
+    Particle &p1 = *bufferParticles[i];
+    const auto pos = p1.getR();
+    const auto boxmin = pos - cutoff;
+    const auto boxmax = pos + cutoff;
+
+    auto p2Iter = container.getRegionIterator(
+        boxmin, boxmax, IteratorBehavior::ownedOrHalo | IteratorBehavior::forceSequential, nullptr);
+    for (; p2Iter.isValid(); ++p2Iter) {
+      Particle &p2 = *p2Iter;
+
+      auto p3Iter = p2Iter;
+      ++p3Iter;
+      for (; p3Iter.isValid(); ++p3Iter) {
+        Particle &p3 = *p3Iter;
+
+        if constexpr (newton3) {
+          f->AoSFunctor(p1, p2, p3, true);
+        } else {
+          if (i < numOwnedBufferParticles) {
+            f->AoSFunctor(p1, p2, p3, false);
+          }
+          // no need to calculate force enacted on a halo
+          if (p2.isOwned()) {
+            f->AoSFunctor(p2, p1, p3, false);
+          }
+          if (p3.isOwned()) {
+            f->AoSFunctor(p3, p1, p2, false);
+          }
+        }
+      }
+    }
+  }
+
+#if SPDLOG_ACTIVE_LEVEL <= SPDLOG_LEVEL_TRACE
+  timerBufferContainerContainer.stop();
+#endif
+
+  AutoPasLog(TRACE, "Timer Buffer <-> Buffer <-> Buffer       : {}", timerBufferBufferBuffer.getTotalTime());
+  AutoPasLog(TRACE, "Timer Buffer <-> Buffer <-> Container    : {}", timerBufferBufferContainer.getTotalTime());
+  AutoPasLog(TRACE, "Timer Buffer <-> Container <-> Container : {}", timerBufferContainerContainer.getTotalTime());
+}
+
 template <typename Particle>
 template <class Functor>
 std::tuple<Configuration, std::unique_ptr<TraversalInterface>, bool> LogicHandler<Particle>::selectConfiguration(
-    Functor &functor) {
+    Functor &functor, const InteractionTypeOption &interactionType) {
   bool stillTuning = false;
   Configuration configuration{};
   std::optional<std::unique_ptr<TraversalInterface>> traversalPtrOpt{};
+  auto *const autoTuner = _autoTunerRefs[interactionType].get();
```

Comment:
I don't understand what you mean, but the suggestion does not work: `'const' qualifier may not be applied to a reference`.

Reply:
Ah my bad, I overlooked the `const`:

```suggestion
  auto &autoTuner = *_autoTunerRefs[interactionType];
```

My point is that you can have a reference here instead of a `const` Pointer, which is (almost) the same.

Reply:
Done in [b2b7443](https://github.com/AutoPas/AutoPas/pull/751/commits/b2b74432c6c22856816a19b13c9e3245c565b9fa)